opentelemetry/src/baggage.rs

```diff
@@ -161,8 +161,8 @@ impl Baggage {
 
     /// Removes a name from the baggage, returning the value
     /// corresponding to the name if the pair was previously in the map.
-    pub fn remove<K: Into<Key>>(&mut self, key: K) -> Option<(Value, BaggageMetadata)> {
-        self.inner.remove(&key.into())
+    pub fn remove<K: AsRef<str>>(&mut self, key: K) -> Option<(Value, BaggageMetadata)> {
```

Comment:
would this be considered user-affecting? Lets include this in the changelog, along with the other Baggage changes.

Reply:
Added a changelog entry, but I'm not sure, if it is really a breaking change. Better be safe than sorry 😅